src/lib/data/governingEntities.ts

```diff
@@ -0,0 +1,107 @@
+import { EntityPrototypeId } from '../../db/models/entityPrototype';
+import { GoverningEntityId } from '../../db/models/governingEntity';
+import { PlanId } from '../../db/models/plan';
+import { Database } from '../../db/type';
+import { InstanceDataOfModel } from '../../db/util/raw-model';
+import { organizeObjectsByUniqueProperty } from '../../util';
+
+/**
+ * A map from governing entity ids to an object that contains both the
+ * governingEntity and governingEntityVersion for each entry.
+ */
+export type MapOfGoverningEntities = Map<
+  GoverningEntityId,
+  {
+    governingEntity: InstanceDataOfModel<Database['governingEntity']>;
+    governingEntityVersion: InstanceDataOfModel<
+      Database['governingEntityVersion']
+    >;
+    customRef: string;
+  }
+>;
+
+export const getAllGoverningEntitiesForPlan = async ({
+  database,
+  planId,
+  prototypes,
+}: {
+  database: Database;
+  planId: PlanId;
+  version: 'latest';
+  /**
+   * A map of prototypes that **must** include the prototype for governing
+   * entities of this plan.
+   *
+   * Usually the full list of prototypes for a plan is passed in.
+   */
+  prototypes: Map<
+    EntityPrototypeId,
+    InstanceDataOfModel<Database['entityPrototype']>
+  >;
+}): Promise<MapOfGoverningEntities> => {
+  const ges = await database.governingEntity.find({
+    where: {
+      planId,
+    },
+  });
+  const gevs = await database.governingEntityVersion.find({
+    where: (builder) =>
+      builder
+        .whereIn('governingEntityId', [...new Set(ges.map((ge) => ge.id))])
```

Comment:
Governing entity IDs already must be unique, due to DB primary key constraints, why put them through `Set`?
```suggestion
        .whereIn('governingEntityId', ges.map((ge) => ge.id))
```